src/Compilers/Core/Portable/CommandLine/CommonCommandLineArguments.cs

```diff
@@ -279,6 +279,40 @@ internal CommandLineArguments()
         {
         }
 
+        internal static string NormalizePathPrefix(string filePath, ImmutableArray<KeyValuePair<string, string>> pathMap)
```

Comment:
Moved from SourceFileResolver. 

I struggled a bit on finding an appropriate place to put this method.  Considered a couple of other places like FileUtilities.cs.  But there didn't seem to be a perfect fit for this method.   Open to suggestions here.